lib/IRGen/GenMeta.cpp

```diff
@@ -3383,8 +3393,9 @@ namespace {
       B.addBitCast(IGM.getDeletedMethodErrorFn(), IGM.FunctionPtrTy);
     }
 
-    void addPlaceholder(MissingMemberDecl *) {
-      llvm_unreachable("cannot generate metadata with placeholders in it");
+    void addPlaceholder(MissingMemberDecl *m) {
+      assert(m->getNumberOfVTableEntries() == 0
```

Comment:
We should not be adding placeholders to parsed classes at all, which is the only kind of class that should show up here; I think you can keep the old assertion

Reply:
If I don't change the assertion, then it crashes on the new MissingMethodDecls I'm creating for the missing stored properties.